### PR TITLE
[EOC-260] [SOCKET] Adding members to a list

### PR DESCRIPTION
**Pull request by alemikolo**
**Adding list member via sockets was refactored.** After this change: adding, deleting, restoring, list's items via socket stopped working (because of problems with two sockets instances - withSocket and socket instance from file `sockets/index.js`). So I had to change a little bit code not related to my task for the application to continue to work correctly.
I had to remove the use of withSocket hoc from List and Comments components and replaced it with the socket from file `sockets/index.js`. An additional change was related to rooms' names. In new methods: joinRoom and leaveRoom we create room name using Routes enum, where we don't have route list but route sack. So I had to change all the rooms' names from `list-${id}` to `sack-${id}`.
I didn't remove socket 's event listeners and other socket s' functionalities related to items and comments. They'll be removed in other tasks related to items.

### CODE REVIEW

File: client/src/modules/cohort/model/actions.js [hunk 1]
```diff
@@ -294,6 +294,7 @@ export const addCohortMember = (cohortId, email, socket) => dispatch =>
     .then(json => {
       if (json._id) {
         const data = { cohortId, json };
+
```
Comment (marekrozmus) [inline client/src/modules/cohort/model/actions.js]: is this needed for this PR?

**Comment by alemikolo** [comment]
Socket logic (regarding adding cohort member) was refactored

File: server/sockets/index.js [hunk 2]
```diff
@@ -60,28 +62,34 @@ const socketListenTo = server => {
...
       socket.join(room);
+      cohortViewClients.set(userId, socket.id);
     });
 
     socket.on('leaveCohortRoom', data => {
-      const { room } = data;
+      const { room, userId } = data;
 
       socket.leave(room);
+      cohortViewClients.set(userId, socket.id);
```
Comment (adamklepacz) [inline server/sockets/index.js]: should be delete on leave?

File: server/sockets/index.js [hunk 3]
```diff
@@ -92,15 +100,16 @@ const socketListenTo = server => {
...
     updateItemState(socket);
     deleteItemWS(socket);
     restoreItemWS(socket);
     updateItemWS(socket);
     addCommentWS(socket);
-    sendListsOnAddCohortMember(socket, dashboardClients);
+    sendListsOnAddCohortMemberWS(socket, dashboardViewClients);
 
-    addCohortMemberWS(socket, cohortsClients);
+    addCohortMemberWS(socket, allCohortsViewClients);
```
Comment (adamklepacz) [inline server/sockets/index.js]: why this empty line in 111?

Comment (alemikolo) [reply server/sockets/index.js]: To separate list's and cohort's listeners

**Pull request merged**